pkg/listwatch/listwatch.go

```diff
@@ -219,15 +219,23 @@ func (mlw *multiListerWatcher) newMultiWatch(resourceVersions map[string]string,
 	}
 
 	mlw.state = watching
-	wg.Add(len(mlw.lwMap)+1)
+	wg.Add(1)
+
+	// result chan must be closed,
+	// once all event sender goroutines exited.
+	go func() {
+		wg.Wait()
+		close(combinedResult)
+		close(result)
+	}()
```

Comment:
Why did you move this up here? If I see correctly, moving this up here makes no difference, since you're just changing the order of running two go routines.

Reply:
Ah, wait. I didn't notice that the `for n, lws := range mlw.lwMap` loop was outside the goroutine...

Reply:
Yeah, I had to move this up, otherwise the `Stop()` wouldn't propagate completely, because nobody's waiting for that WaitGroup